src/reaction_network.jl

```diff
@@ -123,6 +144,32 @@ function coordinate(name, ex::Expr, p, scale_noise)
     expr_arr_to_block(exprs)
 end
 
+# min_reaction_network coordination function, actually does all the work of the macro.
+function min_coordinate(name, ex::Expr, p, scale_noise)
```

Comment:
Or maybe you also indicated some rewriting to facilitate that. Maybe something in this direction:
```julia
function addodes_internal(rn)
    ...
    return :(((f::typeof($rn)))(du, u, p, t::Number) = f.f(du, u, p, t))
end
addodes!(rn) = eval(addodes_internal)
```
Then we have
```julia
macro reaction_network(args...)
    return quote
        internal___var___rn = @min_reaction_network(args)
        $(addodes_internal(internal___var___rn))
        $(addodes_internal(internal___var___rn))
        $(addodes_internal(internal___var___rn))
        internal___var___rn
    end
end
```
On a closer inspection there are a few problem with that code, but something along those lines might work. Would save us having to write all the macros twices as well (that is, having to have two macros for each input combination).

But if we wanted something like this it might be possible.

Reply:
Yeah, there could definitely be some code reuse/refactoring. I didn't want to touch the old macro initially as I'm not super comfortable with that code, and wanted to make sure the basic design is ok with everyone before refactoring anything. `eval`s shouldn't be needed in the old macro, but portions of the shared expression generation could be factored out. i.e.

```
genminnetwork_exprs!
genode_exprs!
gensde_exprs!
genjump_exprs!
```
could be called internally by both the old macro and the new macro/query functions. Since it sounds like the basic design is ok with everyone I could refactor a bit when I have a chance.